### PR TITLE
Fixes #125:  Added A Built-in Functions Page Part-1 in Javascript for a quick revision

### LINKED ISSUE #125: [Feature Request] JavaScript Concepts for Revision (Interview Purpose)
### Is your feature request related to a problem? Please describe.
I'm frustrated when I have an Javascript or frontend interview and I am tired of searching each and every page on google for a better concepts, questions asked in Interview, So I wanted a solution for so long. Now I wanna give my time to compile the best of them. 

### Describe the solution you'd like
I have compiled most asked questions on Javascript and proper articles to understand them. I want to add more conceptual related articles all at one place. I wanna contribute here because it can be very helpful for the community. 

### Describe alternatives you've considered
I will try to link best video explainations if required. 

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by Akshithsaai**
### Description
I have Added A Page for Learning Built-in Functions in JS used in real-time coding Interviews. These concepts are very essential for succeeding a javascript coding interview.

Fixes #125 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please sync your main repo with the latest changes to package.json and package-lock.json. The version of Docusaurus that is currently being used is 3. Please compare your changes with the main repo.

**Comment by Akshithsaai** [comment]
@ajay-dhangar My version of docusaraus is 3.5.2.

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@heroicons/react": "^2.1.5",
     "@mdx-js/react": "^3.0.0",
     "clsx": "^2.0.0",
+    "docusaurus": "^1.14.7",
```
Comment (ajay-dhangar) [inline package.json]: delete line number `27` it's not part of our `package.json`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
If you want to delete or close your PR, you can do so. However, you must first sync your changes with the latest code in our main repository. Also, you cannot push changes to your `package.json` and `package-lock.json` files, as this could be dangerous for our setup.

File: package.json (hunk 1 above)
Comment (Akshithsaai) [reply package.json]: @ajay-dhangar Should I add package.json to gitignore ?

**Comment by ajay-dhangar** [comment]
My suggestion is that you first save your feature's code in a safe place on your machine. Then delete your code and sync the latest main branch code. Make sure you don't edit `package.json` and `package-lock.json`, because this can be dangerous for our setup. If you need to add any packages or dependencies, please share them with us so we can better understand your changes.

**Comment by Akshithsaai** [comment]
Yes @ajay-dhangar , I have made sure to delete my feature changes and pull from main branch so that my local gets updated and then I have added my feature to the local before commiting it. I have pushed same package codes present in main , as I didnt need any additiional packages or dependencies. 

I am new to opensource so It took me many commits to understand this. Please tell me if any additional changes are required.

**Comment by ajay-dhangar** [comment]
> Yes @ajay-dhangar , I have made sure to delete my feature changes and pull from main branch so that my local gets updated and then I have added my feature to the local before commiting it. I have pushed same package codes present in main , as I didnt need any additiional packages or dependencies.
> 
> I am new to opensource so It took me many commits to understand this. Please tell me if any additional changes are required.

Don't worry, my comment was directed at my own repo, not your fork. You can delete your fork or, if you prefer, you can sync and check out which lines your file added and resolve any conflicts that arise.

**Comment by Akshithsaai** [comment]
Ok @ajay-dhangar .

**Comment by Akshithsaai** [comment]
@ajay-dhangar please merge my PR. I think all the conflicts are resolved.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
You need create new branch with ne PR and add code for only `docs/languages/javascript/ex-10.md`

**Pull request closed without merging**